### PR TITLE
[pt2e][quant] Make move_exported_model_to_train/eval idempotent

### PR DESCRIPTION
**Pull request by andrewor14**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #142239

Summary: Before we would recompile the model unnecessarily even
if the model is already in the desired mode. For training
frameworks that assume `model.train()` is idempotent and calls
this before every single training step, this led to a bunch of
tiny graphs and poor performance. This commit makes these calls
no-ops if we're already in the target train/eval mode.

Test Plan:
python test/test_quantization -k TestQuantizePT2E.test_allow_exported_model_train_eval_idempotent

### CODE REVIEW

**Review by jerryzh168 — APPROVED** [review]

File: torch/ao/quantization/pt2e/export_utils.py [hunk 1]
```diff
@@ -195,7 +195,12 @@ def _move_exported_model_to_eval(model: torch.fx.GraphModule):
 
     This is equivalent to model.eval() but only for certain special ops like dropout, batchnorm.
     QAT users should call this before performing inference on the model.
+
+    This call is idempotent; if the model is already in eval mode, nothing will happen.
     """
+    if not model.training:
+        return model
+    model.training = False
```
Comment (jerryzh168) [inline torch/ao/quantization/pt2e/export_utils.py:203]: is this attribute preserved in the transformations after export, will it be a problem if it's not?

Comment (andrewor14) [reply torch/ao/quantization/pt2e/export_utils.py:203]: Hmm good point, maybe safer to use a different private flag
